std/math.d

```diff
@@ -4707,7 +4711,24 @@ real hypot(real x, real y) @safe pure nothrow @nogc
  */
 real ceil(real x) @trusted pure nothrow @nogc
 {
-    version (Win64_DMD_InlineAsm)
+    static real genericCeil(real x) @trusted pure nothrow @nogc
+    {
+        // Special cases.
+        if (isNaN(x) || isInfinity(x))
+            return x;
+
+        real y = floorImpl(x);
+        if (y < x)
+            y += 1.0;
+
+        return y;
+    }
+
+    version(GNU)
+    {
+        return genericCeil(x);
+    }
+    else version (Win64_DMD_InlineAsm)
```

Comment:
Wouldn't this be a problem with the version definition? Based on this change alone, I don't see a reason why this would be necessary.

Reply:
```
version (CRuntime_Microsoft)
{
    version (D_InlineAsm_X86)
        version = MSVC_InlineAsm;
    else version (...)
        version = MSVC_InlineAsm;
}
```
Or whatever suitable name in place of Cruntime_Microsoft.

Reply:
That's indeed much better.